tools/utils/utils.cpp

```diff
@@ -8,32 +8,62 @@
 
 namespace evmc
 {
-bytes from_hex(const std::string& hex)
+namespace
+{
+inline int from_hex_digit(char h)
+{
+    if (h >= '0' && h <= '9')
+        return h - '0';
+    else if (h >= 'a' && h <= 'f')
+        return h - 'a' + 10;
+    else if (h >= 'A' && h <= 'F')
+        return h - 'A' + 10;
+    else
+        throw std::out_of_range{"not a hex digit"};
+}
+
+template <typename OutputIt>
+inline void from_hex(const char* hex, size_t size, OutputIt result)
 {
-    if (hex.length() % 2 == 1)
+    // Omit the optional 0x prefix.
+    if (size >= 2 && hex[0] == '0' && hex[1] == 'x')
```

Comment:
You could do
```
size_t i = ((size >= 2 && hex[0] == '0' && hex[1] == 'x') ? 2 : 0);
```
and still pass `hex` as `const std::string&`

Reply:
Done.